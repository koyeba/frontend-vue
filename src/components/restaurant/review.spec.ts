import { mount } from '@vue/test-utils';
import RestaurantReview from './review.vue';

const mockReview = {
  id: "test-review-1", 
  rating: 4,
  text: "Great restaurant!\n\nI really enjoyed the food."
};

describe(`restaurant-review`, () => {
  test(`no props`, (context) => {
    const wrapper = mount(RestaurantReview, {
      global: { plugins: [context.router] }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`renders a review`, (context) => {
    const wrapper = mount(RestaurantReview, {
      global: { plugins: [context.router] },
      props: { review: mockReview }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
