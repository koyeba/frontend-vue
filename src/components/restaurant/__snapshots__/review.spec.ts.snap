// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`restaurant-review > no props 1`] = `"<!--v-if-->"`;

exports[`restaurant-review > renders a review 1`] = `
"<li class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated va-company-review mb-4\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"opacity: NaN;\\"></div>
      <div class=\\"v-progress-linear__buffer\\" style=\\"opacity: NaN; width: 0%;\\"></div>
      <div class=\\"v-progress-linear__indeterminate\\">
        <div class=\\"v-progress-linear__indeterminate long\\"></div>
        <div class=\\"v-progress-linear__indeterminate short\\"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-card-text\\">
    <div class=\\"mb-3\\">
      <div class=\\"v-rating v-rating--readonly v-theme--light\\"><label for=\\"v-rating-v-0-0\\" class=\\"v-rating__item--full\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
          <!---->
        </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-0\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"0\\">
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-v-0-0-5\\" class=\\"v-rating__item--half\\"><span class=\\"v-rating__hidden\\">Rating 0.5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 0.5 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-0-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"0.5\\"><label for=\\"v-rating-v-0-1\\" class=\\"v-rating__item--full\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 1 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-1\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"1\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-v-0-1-5\\" class=\\"v-rating__item--half\\"><span class=\\"v-rating__hidden\\">Rating 1.5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 1.5 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-1-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"1.5\\"><label for=\\"v-rating-v-0-2\\" class=\\"v-rating__item--full\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 2 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-2\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"2\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-v-0-2-5\\" class=\\"v-rating__item--half\\"><span class=\\"v-rating__hidden\\">Rating 2.5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 2.5 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-2-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"2.5\\"><label for=\\"v-rating-v-0-3\\" class=\\"v-rating__item--full\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 3 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-3\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"3\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-v-0-3-5\\" class=\\"v-rating__item--half\\"><span class=\\"v-rating__hidden\\">Rating 3.5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 3.5 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-3-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"3.5\\"><label for=\\"v-rating-v-0-4\\" class=\\"v-rating__item--full\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 4 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-4\\" type=\\"radio\\" checked=\\"\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"4\\"></div>
          <!---->
        </div>
        <div class=\\"v-rating__wrapper\\">
          <!---->
          <div class=\\"v-rating__item\\"><label for=\\"v-rating-v-0-4-5\\" class=\\"v-rating__item--half\\"><span class=\\"v-rating__hidden\\">Rating 4.5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 4.5 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-4-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"4.5\\"><label for=\\"v-rating-v-0-5\\" class=\\"v-rating__item--full\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light text-warning v-btn--density-compact v-btn--size-small v-btn--variant-plain\\" aria-label=\\"Rating 5 of 5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
                <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
                <!---->
                <!---->
              </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-v-0\\" id=\\"v-rating-v-0-5\\" type=\\"radio\\" tabindex=\\"-1\\" readonly=\\"\\" value=\\"5\\"></div>
          <!---->
        </div>
      </div>
    </div>
    <p class=\\"text-body-2 mb-0\\" style=\\"white-space: pre-line;\\">Great restaurant!

      I really enjoyed the food.</p>
  </div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</li>"
`;
